lib/model-definition.js

```diff
@@ -238,9 +238,17 @@ ModelDefinition.prototype.build = function(forceRebuild) {
       };
       if (typeof prop === 'object' && prop !== null) {
         for (const a in prop) {
+          const isNestedArrayProp = Array.isArray(prop[a]);
           // Skip the type property but don't delete it Model.extend() shares same instances of the properties from the base class
-          if (a !== 'type') {
+          if (a !== 'type' && !isNestedArrayProp) {
```

Comment:
I have landed a PR handling a condition like this - https://github.com/strongloop/loopback-datasource-juggler/pull/1699.

It makes me wonder if some architectural optimization can be done around this structure, instead of checking if a structure is an array within an array (including my implementation).

@bajtos @raymondfeng

Reply:
+1 to find a way how to address this problem in a wider context.

I am proposing the following:
- Land this PR first, to fix the immediate problem
- Create a follow-up spike story to investigate better solutions. Time-box it to 2-3 days only. I don't want to spend too much time on improving this legacy codebase, we are likely to throw most of it away as part of juggler refactor in the future.

Reply:
Actually, I've talked to @raymondfeng and have concluded to revert this change. Right now the following two flavours of an array property work:
```javascript
   const model = db.define('test', {
      randomReview: {
        type: [
          {
            type: 'string',
          },
        ],
        mongodb: {
          dataType: 'Decimal128',
        },
      },
    });
```
or
```javascript
   const model = db.define('test', {
      randomReview: {
        type: [String ],
        mongodb: {
          dataType: 'Decimal128',
        },
      },
    });
```